keypatch.py

```diff
@@ -310,33 +311,48 @@ def __init__(self, arch=None, mode=None):
     def get_hardware_mode():
         (arch, mode) = (None, None)
 
-        # heuristically detect hardware setup
-        info = idaapi.get_inf_structure()
+        try:
```

Comment:
Plz change all `try except` to `idaapi.IDA_SDK_VERSION >= 900` like code, thanks!

Reply:
@kabeor , done!